### ISSUE TITLE
Improve Scheduler Step Call Positions

### ISSUE THREAD

**Issue by ramppdev** [autrainer/autrainer#17]
Currently schedulers are treated differently for step- and epoch-based training:
- epoch-based training: `scheduler.step()` is called after each epoch / full pass over the dataset
- step-based training: `scheduler.step()` is called after each optimization step / batch passed to the model

This has to be unified by calling the scheduler at the same position during training.

Also, schedulers like `StepLR` it is common to decrease the LR after a full pass over the dataset, others need to modify the LR after each optimization step.
Therefore it is useful to add a new argument like `update_frequency` to the scheduler configuration that specifies if it should be called after each `batch` or after a full `pass` over the dataset (epoch for epoch-based, steps of eval frequency for step-based training).
